### PR TITLE
Support bank statement for Santander Consumer Bank GmbH

### PR DESCRIPTION
**Pull request by auchri**
Add support for the PDF import of the bank statement of [Santander Consumer Bank GmbH](https://www.santanderconsumer.at/).

I am not sure if we should create a second PDF importer for the Austrian Santander Consumer Bank GmbH or do we leave it as it is now.
I have renamed the pdf importer from SantanderConsumerBankAGPDFExtractor to SantanderConsumerBankPDFExtractor to reflect the changes.

If any changes are required, please leave a comment and do not create a "Improve .. PDF-Importer" PR afterwards :)

### CODE REVIEW

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SantanderConsumerBankPDFExtractor.java [hunk 1]
```diff
@@ -185,4 +193,69 @@ private <T extends Transaction<?>> void addFeesSectionsTransaction(T transaction
                 .match("^Fremde Abwicklungsgeb.hr .* (?<fee>[.,\\d]+)\\- (?<currency>[\\w]{3})$")
                 .assign((t, v) -> processFeeEntries(t, v, type));
     }
+
+    private void addKontoauszugDocument()
```
Comment (Nirus2000) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SantanderConsumerBankPDFExtractor.java:197]: We should stick to a standard here, which is also stored in the [Contributing rules](https://github.com/Nirus2000/portfolio/blob/master/CONTRIBUTING.md#auxiliary-classes). It is again a variant for one and the same thing, the reading of account or deposit statements.
Again and again other variants, means always more maintenance effort.
Please change to the variant of the DKB Importer with the defined variables and descriptions.

**Comment by Nirus2000** [comment]
> I am not sure if we should create a second PDF importer for the Austrian Santander Consumer Bank GmbH or do we leave it as it is now.
I have renamed the pdf importer from SantanderConsumerBankAGPDFExtractor to SantanderConsumerBankPDFExtractor to reflect the changes.

I think that's okay... 👍🏻 ... nice idea!

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SantanderConsumerBankPDFExtractor.java (hunk 1 above)
Comment (auchri) [reply name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SantanderConsumerBankPDFExtractor.java]: Can you please have a look now? I think now it should fit

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SantanderConsumerBankPDFExtractor.java [hunk 2]
```diff
@@ -206,56 +207,54 @@ private void addKontoauszugDocument()
         });
         this.addDocumentTyp(type);
 
-        addKontoauszugTaxesTransaction(type);
-        addKontoauszugInterestTransaction(type);
-        addKontoauszugDepositTransaction(type);
-    }
+        // 19.05.2023 19.05.2023 34,00 34,01
+        final String accountStatementBaseRegex = "^(?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) ([\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) (?<amount>[\\.,\\d-]+) ([\\.,\\d-]+) ";
```
Comment (Nirus2000) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SantanderConsumerBankPDFExtractor.java:211]: please adjust this structure to all the other pdf importer.... fintech, dab, comdirec, dkb... ... ...

**Comment by auchri** [comment]
Are the tests ok now?

**Comment by Nirus2000** [comment]
Hello @auchri 
I visualize the difference as an example.
grafik

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/santanderconsumerbank/SantanderConsumerBankPDFExtractorTest.java [hunk 3]
```diff
@@ -256,4 +267,39 @@ public void testDividende02()
...
+
+        assertThat(errors, empty());
+        assertThat(countSecurities(results), is(0L));
+        assertThat(countBuySell(results), is(0L));
+        assertThat(countAccountTransactions(results), is(4L));
+        assertThat(results.size(), is(4));
+        new AssertImportActions().check(results, CurrencyUnit.EUR);
+        
+        // assert transaction
+        assertThat(results, hasItem(taxes(hasDate("2023-05-31"), hasAmount("EUR", 1.66), //
```
Comment (Nirus2000) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/santanderconsumerbank/SantanderConsumerBankPDFExtractorTest.java:288]: hasNote(null)

check all .... we work against the tests not the test decides the result.

**Comment by auchri** [comment]
You said that I should look at the DKB extractor for an example. 

So I looked at the [addAccountStatementTransaction()](https://github.com/buchen/portfolio/blob/d902d32a47f4bff3da88dccfc4f5bcea9165d5eb/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DkbPDFExtractor.java#L551) method and adjusted my code accordingly. 

Maybe someone should create an example pdf extractor that is up-to-date and is created exactly with your desired structure... code says more than 1000 words.

So which method/pdf importer has the "correct" structure?

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SantanderConsumerBankPDFExtractor.java [hunk 4]
```diff
@@ -185,4 +192,98 @@ private <T extends Transaction<?>> void addFeesSectionsTransaction(T transaction
                 .match("^Fremde Abwicklungsgeb.hr .* (?<fee>[.,\\d]+)\\- (?<currency>[\\w]{3})$")
                 .assign((t, v) -> processFeeEntries(t, v, type));
     }
+
```
Comment (Nirus2000) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SantanderConsumerBankPDFExtractor.java:195]: please shift the function about the treatment of taxes and fees (to line 168...)

Structure: [Contruting rules](https://github.com/buchen/portfolio/blob/master/CONTRIBUTING.md#anatomy-of-a-pdf-importer)

- Client + Importer label
- Transaction types
- Taxes and fees 
- other helping classes

All other importers are set up like this.
Simple task for you... and quickly done. 👍🏻

**Comment by buchen** [comment]
rebased and merged - thanks @auchri

**Pull request closed without merging**